onnx/defs/generator/defs.cc

```diff
@@ -215,18 +220,28 @@ ONNX_OPERATOR_SET_SCHEMA(
              "tensor(bool)"},
             "Constrain output types. Strings and complex are not supported.")
         .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          // Type inference
           if (ctx.getAttribute("dtype") != nullptr) {
             propagateElemTypeFromAttributeToOutput(ctx, "dtype", 0);
           } else {
             propagateElemTypeFromInputToOutput(ctx, 0, 0);
           }
+
+          // Shape inference
           if (hasInputShape(ctx, 0)) {
             auto& input_shape = getInputShape(ctx, 0);
             if (input_shape.dim_size() != 2) {
               fail_shape_inference("Input tensor must be 2-dimensional");
             }
+            propagateShapeFromInputToOutput(ctx, 0, 0);
+          } else {
+          // Rank inference
+            auto* output_shape =
```

Comment:
E.g., with the utility mentioned above, this could be just "setOutputRank(ctx, 0, 2);"